src/map/utils/battleutils.cpp

```diff
@@ -4368,15 +4298,25 @@ namespace battleutils
                 {
                     for (std::size_t i = 0; i < a->members.size(); ++i)
```

Comment:
Please use range-based-for here, unless you have a very solid reason for needing an index into a->members

Reply:
I can't comment on the line that has it, but while you're in here and understand what's going on, can you rename the `a` in `for (auto& a : taUser->PParty->m_PAlliance->partyList)` to something a bit more descriptive? Thanks!

Reply:
Haha sure. I'll handle the rest of these comments and test it again. Thanks for the feedback!

Reply:
Haven't tested it but while I'm in here how do you feel about compressing this into one loop:

```
            std::vector<CParty*> taPartyList;
            if (taUser->PParty->m_PAlliance != nullptr)
            {
                taPartyList = taUser->PParty->m_PAlliance->partyList;
            }
            else
            {
                taPartyList.emplace_back(taUser->PParty);
            }
            for (auto&& party : taPartyList)
            {
                for (auto&& member : party->members)
                {
                        ...
```

Reply:
Sure, as long as it's tested and it works, and doesn't introduce any fuckery